src/commonMain/kotlin/exchange.dydx.abacus/output/Account.kt

```diff
@@ -1819,24 +1820,33 @@ data class TradingRewards(
 
                             (comparison == ComparisonOrder.descending) -> {
                                 // item is older than obj
-                                addHistoricalTradingRewards(result, obj, period, lastStart)
-                                result.add(obj)
+                                val modified = mapOf(
```

Comment:
hm so in this path, because `item` is older than `obj`, we actually want to add `obj` (not `item`). *This is why only `objIndex` is getting incremented (not `dataIndex`). This whole blob of code is basically just trying to merge in `data` (`item`) and `objs` in chronological order (newest -> oldest). LMK if this would be helpful to talk through live! I had to read this like 5 times to understand it lol

Reply:
Oh I see item is older than obj in this path. Thanks for explaining!